### ISSUE TITLE
Compound launch config no longer works

### ISSUE THREAD

**Issue by DanTup** [Microsoft/vscode#35425]
I don't know when this started failing as it's been a while since I last used it. I have a compound launch config to allow debugging both my extension and its debug adapter:

Extension + Dart Debug Server
https://github.com/Dart-Code/Dart-Code/blob/d5f1a21354a35f17e9b5d716fe7132cdaa2e2d48/.vscode/launch.json#L66

Today (latest stable Code) when I press F5 with this debug config selected, I get this message after 10 seconds in the extension host:

> Extension host did not start in 10 seconds, it might be stopped on the first line and needs a debugger to continue.

The debug output in the main Code instance shows:

```text
Debugging with inspector protocol because Node.js v8.6.0 was detected.
/Applications/Visual Studio Code.app/Contents/MacOS/Electron --debugBrkPluginHost=26962 --debugId=e496bb00-d60a-4bd3-9c28-a4321c250679 --extensionDevelopmentPath=/Users/danny/Dev/Dart-Code 
node --inspect=14925 --debug-brk out/src/debug/dart_debug_entry.js --server=4711 
Debugger listening on ws://127.0.0.1:14925/a6b185f3-717c-4b23-a48f-b9492ae5876c
Debugger attached.
waiting for debug protocol on port 4711
```

The dev tools in the main Code instance show this:

> (ERROR): Cannot read property 'windows' of undefined: TypeError: Cannot read property 'windows' of undefined

**Comment by isidorn** [comment]
@DanTup can you please try reproducing this with the latest vscode insiders?

**Comment by DanTup** [comment]
@isidorn In Insiders, I'm not getting the "failed to connect within 10s" message and it seems like both debug sessions are active, however it doesn't seem like I'm able to hit any breakpoints in the debug adapter either (for example, I put a breakpoint in `evaluateRequest` and I'm hovering variables and seeing tooltips, but without breaking).

**Comment by isidorn** [comment]
@DanTup ok thanks for following up, let's focus on that issue, since the one mentioned here seems to be fixed.

1. Can you reproduce that issue if you do not use compound debugging?
2. Do breakpoints appear as verified?
3. Can you still evalute things in the repl, or is it just breakpoints not being hit?

Take note that we are now using node2 debug adapter to debugging extensions so also including @roblourens as he might have additional questions

**Comment by DanTup** [comment]
@isidorn I'm not really sure how to launch without using the compound config, since if I only select the debugger server, then the extension host doesn't launch when I run? (and if I only select the extension then obviously it definitely won't hit breakpoints in the debug adapter).

**Comment by DanTup** [comment]
Actually, I can probably just run the extension without debugging before-hand.. I'm give it a go - might not be till tomorrow now though, away from machine.

**Comment by isidorn** [comment]
@DanTup sure, let me know once you have some details. No rush

**Comment by DanTup** [comment]
@isidorn Ok, good news - I'm just a noob!

Turns out, this works fine in Insiders. While debugging the original issue, I'd commented out `debugServer` from the `launch.json` in my sample app. So when I used Insiders for testing, it wasn't connecting to the debug server I'd launched and I hadn't noticed. With that put back, it is correctly hitting breakpoints both using the compound config and without.

So, if the original bug has been fixed (it seems like it - but I don't know if there's something else affecting my testing of insiders vs stable) then I believe there's no other issues to resolve :-)
